src/control/server/mgmt_system.go

```diff
@@ -303,18 +310,24 @@ func (svc *mgmtSvc) join(ctx context.Context, req *batchJoinRequest) *batchJoinR
 	return resp
 }
 
-// reqGroupUpdate requests an asynchronous group update.
-func (svc *mgmtSvc) reqGroupUpdate(ctx context.Context) {
+// reqGroupUpdate requests a group update.
+func (svc *mgmtSvc) reqGroupUpdate(ctx context.Context, sync bool) {
 	select {
 	case <-ctx.Done():
-	case svc.groupUpdateReqs <- struct{}{}:
+	case svc.groupUpdateReqs <- sync:
 	}
 }
 
 // doGroupUpdate performs a synchronous group update.
 // NB: This method must not be called concurrently, as out-of-order
 // group updates may trigger engine assertions.
-func (svc *mgmtSvc) doGroupUpdate(ctx context.Context) error {
+func (svc *mgmtSvc) doGroupUpdate(ctx context.Context, forced bool) error {
+	if forced {
+		if err := svc.sysdb.IncMapVer(); err != nil {
```

Comment:
why the map version only incremented when forced?

Reply:
It's automatically incremented any time a member is updated, or when this special path is followed after a leadership change. In the latter case, we can't guarantee that the map version persisted in the raft db is greater than the last update, and so we force an increment to the version before performing the update. This forced map update is necessary to avoid some OOG issues seen after leadership changes.